### PR TITLE
Feature/login as a user

### PR DESCRIPTION
**Pull request by guerretta16**
#3 
- RTK Api added for `Quote Hub`.
- Constants added for the URL and API_KEY `Quote Hub`.
- Types were added for `Quote Hub` response and `User` login form.
- Component Popup added to show the Login form.
- Component Login Form added.
   - Validates the username or email and password.
   - Execute 3 functions: `Get csrf token`, `Get access token` and `Get auth user info`.
- Refactor form Header. Popup component added there and if `access_token` exists, login button will hidde.
- Token slice was added with persist.
- Store was refactored to persist the info just for the `token`.

### RTK Endpoints

- Get csrf token. Used to protect the comunication with server.
- Get access token. (Login)
- Get auth user info.

### CODE REVIEW

File: src/constants/quote_hub/quote_hub.constants.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const QUOTE_HUB_API = {
+    API_KEY: import.meta.env.VITE_QUOTE_HUB_API_URL,
+    URL: 'http://localhost:8000/api/'
```
Comment (f-esquivel) [inline src/constants/quote_hub/quote_hub.constants.ts]: - [x] The `Quote Hub` API doesn't receive an `API_KEY` parameter.
- [x] The `URL`  constant should be a environment variable 😅

File: src/api/rtk/quote_hub.api.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                url: 'auth/me',
+                method: 'GET',
+                headers: {
+                    'Authorization': 'Bearer ' + token
+                }
+            })
+
+        })
+    })
+});
```
Comment (f-esquivel) [inline src/api/rtk/quote_hub.api.ts]: Refactor the `Quote Hub` RTK implementation.

More instructions on call.

File: src/api/rtk/quote_hub.api.ts
Comment (f-esquivel) [inline src/api/rtk/quote_hub.api.ts:1]: `quoteHub.api.ts`

Use camel case.

File: src/app/features/token/tokenSlice.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+interface TokenState {
+    token: string;
+}
+
+// Define the initial state using that type
+const initialState: TokenState = {
+    token: ''
+};
+
+export const tokenSlice = createSlice({
```
Comment (f-esquivel) [inline src/app/features/token/tokenSlice.ts]:
```suggestion
export const authSlice = createSlice({
```

Refactor all the files to use this name instead.

File: src/components/LoginForm/LoginForm.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import { setToken }                                                                          from '@/app';
+import { useAppDispatch }                                                                    from '@/hooks';
+import { UserLogin }                                                                         from '@/types';
+import './LoginForm.css';
+
+
+export const LoginForm = () => {
+
+    const [getAccessToken, { isLoading: loadingAccessToken }] = useLazyGetAccessTokenQuery();
+    const [getCsrfToken, { isLoading: loadingCsrfToken }] = useLazyGetCsrfTokenQuery();
```
Comment (f-esquivel) [inline src/components/LoginForm/LoginForm.tsx]: Call this query on a top level component like the main layout. All the API queries are going to need this variable.

File: src/components/LoginForm/LoginForm.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import { useLazyGetAccessTokenQuery, useLazyGetAuthUserDataQuery, useLazyGetCsrfTokenQuery } from '@/api';
+import { setToken }                                                                          from '@/app';
+import { useAppDispatch }                                                                    from '@/hooks';
+import { UserLogin }                                                                         from '@/types';
+import './LoginForm.css';
+
+
+export const LoginForm = () => {
+
+    const [getAccessToken, { isLoading: loadingAccessToken }] = useLazyGetAccessTokenQuery();
```
Comment (f-esquivel) [inline src/components/LoginForm/LoginForm.tsx]: Move to top level component.

File: src/components/LoginForm/LoginForm.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    const [getCsrfToken, { isLoading: loadingCsrfToken }] = useLazyGetCsrfTokenQuery();
+    const [getAuthUserInfo] = useLazyGetAuthUserDataQuery();
```
Comment (f-esquivel) [inline src/components/LoginForm/LoginForm.tsx]: Top level component

File: src/constants/quote_hub/quote_hub.constants.ts
Comment (f-esquivel) [inline src/constants/quote_hub/quote_hub.constants.ts:1]: `quoteHub.constants.ts` Use camel case.

File: src/types/entities/QuoteHub.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+export type AccessToken = {
+    token: string
+}
+
+export type Categories = {
+    id: string
```
Comment (f-esquivel) [inline src/types/entities/QuoteHub.ts:9]: type id correctly with `CategoryType`

File: src/types/entities/index.ts [hunk 8]
```diff
@@ -1 +1,5 @@
-export * from './ApiNinjasQuote';
\ No newline at end of file
+export * from './ApiNinjasQuote';
+export * from './User';
+export * from './QuoteHub';
+export * from './User';
+export * from './QuoteHub';
```
Comment (f-esquivel) [inline src/types/entities/index.ts]: This doesn't throw error? 🤔

File: src/types/entities/User/User.ts [hunk 9]
```diff
@@ -0,0 +1,4 @@
+export interface UserLogin {
+    user_identifier: string;
+    password: string;
+}
```
Comment (f-esquivel) [inline src/types/entities/User/User.ts]: Move this to a forms payloads types structure.

**Review by f-esquivel — CHANGES_REQUESTED** [review]

File: src/types/entities/QuoteHub.ts (hunk 7 above)
Comment (guerretta16) [reply src/types/entities/QuoteHub.ts:9]: This id is for the DB info, for the relation between User and Category.
I just change the name, which should be a CategoryType.

**Review by f-esquivel — COMMENTED** [review]
🤠👌🏾

**Review by f-esquivel — APPROVED** [review]

**Pull request merged**